crates/engine/local/src/payload.rs

```diff
@@ -47,12 +47,22 @@ where
 impl<ChainSpec> PayloadAttributesBuilder<op_alloy_rpc_types_engine::OpPayloadAttributes>
     for LocalPayloadAttributesBuilder<ChainSpec>
 where
-    ChainSpec: Send + Sync + EthereumHardforks + 'static,
+    ChainSpec: Send + Sync + reth_chainspec::EthChainSpec + EthereumHardforks + 'static,
 {
     fn build(&self, timestamp: u64) -> op_alloy_rpc_types_engine::OpPayloadAttributes {
+        // Add dummy system transaction if running on the `dev` chain
+        let transactions = if self.chain_spec.chain() == alloy_chains::Chain::dev() {
```

Comment:
I think we can remove this check here in case the user runs --dev with a modified chain id

Reply:
Good point but what's the best way to find if the node is running in dev mode then, considering that if we keep this dummy transaction here then it would always be added in the block even if the node is not running in the dev mode, ref https://github.com/paradigmxyz/reth/blob/b6875b13af2f4cb754d6ea0811a3681088fc3518/crates/optimism/payload/src/payload.rs#L68, please correct me if I'm wrong

Reply:
here this is not possible but we only use this type here if in dev mode so we can just make the assumption

with #14064 we can solve this properly

Reply:
Got it, resolved